### PR TITLE
Frontend - backend communication unification in single/cluster backend mode

### LINKED ISSUE #514: Unify the way how frontend checks whether backend is online in case of single vs. cluster mode
Please unify the way how frontend checks whether backend is online. So even in case of single backend, RMI is not used for checking whether backend is online, but DB is checked instead. (as in cluster mode)

Expected changes for frontend:

* No need to define whether frontend works in single/cluster mode using backend.cluster.mode
* status update whether backend is online will not be instant, but will have certain delay based on the settings of the param backend.alive.limit (I suggest by default 10s)
* when pipeline is run/debug from frontend, it may take up to 2s before certain backend really runs the pipeline

Expected changes for backend:

* backend.id is mandatory also in cluster mode
* backend.cluster.mode property not needed
This update may be also solved in a separate pull request (if not doable before vacation)

### PR DESCRIPTION
**Pull request by eea03**
Resolves #514 

Now frontend always checks if backend is online via database, not RMI.
Furthermore, frontend nor backend does not need to know if backend runs in cluster mode.

### CODE REVIEW

File: backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java [hunk 1]
```diff
@@ -136,21 +135,12 @@
...
     public void init() {
-        try {
-            this.clusterMode = this.appConfig.getBoolean(ConfigProperty.BACKEND_CLUSTER_MODE);
-        } catch (MissingConfigPropertyException e) {
-            LOG.info("Running in single mode because cluster mode property is missing in config.properties, {}", e.getLocalizedMessage());
-        }
-        if (this.clusterMode) {
-            this.backendID = this.appConfig.getString(ConfigProperty.BACKEND_ID);
-        }
+        this.backendID = this.appConfig.getString(ConfigProperty.BACKEND_ID);
```
Comment (tomas-knap) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java]: Note for tomas-knap: what happens if the config.properties BACKEND_ID is missing?

File: backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/scheduling/Scheduler.java [hunk 2]
```diff
@@ -49,37 +48,26 @@
...
     private void init() {
-        try {
-            this.clusterMode = this.appConfig.getBoolean(ConfigProperty.BACKEND_CLUSTER_MODE);
-        } catch (MissingConfigPropertyException e) {
-            LOG.info("Running in single mode because cluster mode property is missing in config.properties, {}", e.getLocalizedMessage());
-        }
-        if (this.clusterMode) {
-            this.backendID = this.appConfig.getString(ConfigProperty.BACKEND_ID);
-        }
+        this.backendID = this.appConfig.getString(ConfigProperty.BACKEND_ID);
```
Comment (tomas-knap) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/scheduling/Scheduler.java]: Note for tomas-knap: what happens if the config.properties BACKEND_ID is missing?

File: backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java (hunk 1 above)
Comment (eea03) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java]: Backend process will not even start without this property and error messages is shown in logs:
`Invocation of init method failed; nested exception is cz.cuni.mff.xrg.odcs.commons.app.conf.MissingConfigPropertyException: Config is missing property: backend.id.`

This is the standard behavior for mandatory parameters after error handling in backend was fixed

Comment (tomas-knap) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java]: Checked, OK.

**Pull request merged**